test/openssl/test_ssl.rb

```diff
@@ -1375,6 +1375,37 @@ def test_minmax_version
     }
   end
 
+  def test_minmax_version_system_default
+    omit "LibreSSL does not support OPENSSL_CONF" if libressl?
+
+    Tempfile.create("openssl.cnf") { |f|
+      f.puts(<<~EOF)
+        openssl_conf = default_conf
+        [default_conf]
+        ssl_conf = ssl_sect
+        [ssl_sect]
+        system_default = ssl_default_sect
+        [ssl_default_sect]
+        MaxProtocol = TLSv1.2
+      EOF
+      f.close
+
+      start_server(ignore_listener_error: true) do |port|
+        assert_separately([{ "OPENSSL_CONF" => f.path }, "-ropenssl", "-", port.to_s], <<~"end;")
+          sock = TCPSocket.new("127.0.0.1", ARGV[0].to_i)
+          ctx = OpenSSL::SSL::SSLContext.new
+          ctx.min_version = OpenSSL::SSL::TLS1_2_VERSION
+          ssl = OpenSSL::SSL::SSLSocket.new(sock, ctx)
+          ssl.sync_close = true
+          ssl.connect
+          assert_equal("TLSv1.2", ssl.ssl_version)
+          ssl.puts("abc"); assert_equal("abc\n", ssl.gets)
+          ssl.close
+        end;
+      end
+    }
+  end
+
```

Comment:
The test looks great!

I want to see you add one more test here with the following conditions to test `ctx.min_version = nil` (respecting `MinProtocol = TLSv1.3`) and `ctx.max_version = OpenSSL::SSL::TLS1_3_VERSION`.

```
[ssl_default_sect]
MinProtocol = TLSv1.3
```

```
ctx.min_version = nil
ctx.max_version = OpenSSL::SSL::TLS1_3_VERSION
...
assert_equal("TLSv1.3", ssl.ssl_version)
```

Reply:
`ctx.min_version = nil` will not make it respect the `MinProtocol` value in the config file, but it will reset the minimum bound to "any version" by calling `SSL_CTX_set_min_proto_version(ctx, 0)`. This PR doesn't change this.

I agree an additional test case would be useful.

Reply:
I see. Thanks for your explanation. So, I assume that if we execute the `ctx.min_version = nil`, the content of the setting `MinProtocol = TLSv1.3` is also gone, right?

Reply:
> I see. Thanks for your explanation. So, I assume that if we execute the `ctx.min_version = nil`, the content of the setting `MinProtocol = TLSv1.3` is also gone, right?

All right. In the additional test, I can see the `ctx.max_version = nil` (or `ctx.min_version = nil`) resets the content of the `MaxProtocol = Foo` (or `MinProtocol = Bar`)

Reply:
That is correct. Since protocol version negotiation chooses the highest common version, the difference isn't observable. It would negotiate TLS 1.3 whether you have `MinProtocol` or not.

I just pushed a new commit with updated tests. In the latter block, `ctx.max_version = nil` negates `MaxProtocol = TLSv1.2` in the config to allow TLS 1.3 connection, as opposed to the first one.

Reply:
All right. The tests look good to me!

Reply:
Thank you for the review!